service.rb

```diff
@@ -26,6 +26,7 @@ def initialize(port)
           response_message << "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
         end
         response_message << "</ul>\n"
+        response_message << daily_data_form
```

Comment:
hmm several reasons: 
- so it can be used elsewhere e.g. in the POST 
- there is no parameter we need to care about in the string (yet)
- it seems isolated enough of a concept to extract

Reply:
Makes sense, thanks. Depending on what you decide to do elsewhere you may end up not needing to use it in more than one place, so I guess we’ll see how it ends up, but there’s no harm in extracting it.